apps/projects/app/components/Shared/FilterBar/FilterBar.js

```diff
@@ -136,24 +275,42 @@ const FilterBar = ({
 
   // Complete list of sorters for DropDown. Parent has only one item, to perform actual sorting.
   const [ sortBy, setSortBy ] = useState('Newest')
-
   const [ textFilter, setTextFilter ] = useState('')
-  const { layoutName } = useLayout()
   const [ sortMenuVisible, setSortMenuVisible ] = useState(false)
   const [ actionsMenuVisible, setActionsMenuVisible ] = useState(false)
-  const [ searchVisible, setSearchVisible ] = useState(false)
-  const { curateIssues, allocateBounty } = usePanelManagement()
+  const [ textFilterVisible, setTextFilterVisible ] = useState(false)
+  const [ filtersDisplayNumber, setFiltersDisplayNumber ] = useState(10)
   const theme = useTheme()
   const actionsOpener = useRef(null)
   const sortersOpener = useRef(null)
-  const searchOpener = useRef(null)
+  const textFilterOpener = useRef(null)
+  const mainFBRef = useRef(null)
+  const rightFBRef = useRef(null)
   const activeFilters = () => {
     let count = 0
     const types = [ 'projects', 'labels', 'milestones', 'statuses' ]
     types.forEach(t => count += Object.keys(filters[t]).length)
     return count
   }
 
+  const recalculateFiltersDisplayNumber = useCallback(() => {
+    const total = mainFBRef.current ? mainFBRef.current.offsetWidth : 0
+    const right = rightFBRef.current ? rightFBRef.current.offsetWidth : 0
+    // 80px is "selectAll" checkbox + padding, etc
+    // width is calculated from total width of main FB div and right FB div
+    // (containing actions, text filter and sorters)
+    const width = total - (right + 120)
+    setFiltersDisplayNumber(Math.floor(width / (128+8)))
```

Comment:
`128+8` – why not `136`? :smile:

Also, I think the comment above about `80px` might be outdated? I'm not sure it's useful to leave in at all, tbh. I think the next person to work with this code will probably already respect the amount of trial-and-error/poking at it involved in this sort of style math.

Reply:
True. It's actually more scientific than simple poking ;) And I'd hope the next time this code is improved will make the math unnecessary.